airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py

```diff
@@ -157,85 +143,79 @@ def get_stream_params(self) -> Mapping[str, Any]:
-        start_date = self.start_date
+        start_date = self.start_date  # type: ignore[attr-defined]
         cursor_value = None
 
-        if stream_state and self.cursor_field and self.cursor_field in stream_state:
+        if stream_state and self.cursor_field and self.cursor_field in stream_state:  # type: ignore[attr-defined]
             # Remove time part from state because API accept 'from_date' param in date format only ('YYYY-MM-DD')
             # It also means that sync returns duplicated entries for the date from the state (date range is inclusive)
-            cursor_value = stream_state[self.cursor_field]
-            stream_state_date = pendulum.parse(stream_state[self.cursor_field]).date()
+            cursor_value = stream_state[self.cursor_field]  # type: ignore[attr-defined]
+            stream_state_date = pendulum.parse(stream_state[self.cursor_field]).date()  # type: ignore[attr-defined]
             start_date = max(start_date, stream_state_date)
 
         # move start_date back <attribution_window> days to sync data since that time as well
-        start_date = start_date - timedelta(days=self.attribution_window)
+        start_date = start_date - timedelta(days=self.attribution_window)  # type: ignore[attr-defined]
 
         # end_date cannot be later than today
-        end_date = min(self.end_date, pendulum.today(tz=self.project_timezone).date())
+        end_date = min(self.end_date, pendulum.today(tz=self.project_timezone).date())  # type: ignore[attr-defined]
 
         while start_date <= end_date:
             if self._timezone_mismatch:
                 return
-            current_end_date = start_date + timedelta(days=self.date_window_size - 1)  # -1 is needed because dates are inclusive
+            current_end_date = start_date + timedelta(days=self.date_window_size - 1)  # type: ignore[attr-defined] # -1 is needed because dates are inclusive
             stream_slice = {
                 "start_date": str(start_date),
                 "end_date": str(min(current_end_date, end_date)),
             }
             if cursor_value:
-                stream_slice[self.cursor_field] = cursor_value
+                stream_slice[self.cursor_field] = cursor_value  # type: ignore[attr-defined]
             yield stream_slice
             # add 1 additional day because date range is inclusive
             start_date = current_end_date + timedelta(days=1)
 
     def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> MutableMapping[str, Any]:
-        params = super().request_params(stream_state, stream_slice, next_page_token)
+        params = super().request_params(stream_state, stream_slice, next_page_token)  # type: ignore[misc]
```

Comment:
for any mypy ignore w/ misc, we should add a comment for why we're ignoring this since it's not obvious why we couldn't fix this